src/config.tsx

```diff
@@ -44,19 +44,23 @@ const { Provider } = configContext;
 
 export type AreaFilter = (value: Area, index: number, array: Area[]) => boolean;
 export type AreaMapper = (value: Area, index: number, array: Area[]) => Area;
+export type AreaSorter = (a: Area, b: Area) => number;
 const defaultAreaFilter: AreaFilter = () => true;
 const defaultAreaMapper: AreaMapper = (area) => area;
+const defaultAreaSorter: AreaSorter = () => -1;
```

Comment:
It looks like this default sorter will change the order in previous versions. I would prefer a compatible behavior. Could you keep the same order when sorter is not provided?

<img width="399" alt="image" src="https://user-images.githubusercontent.com/26411634/180434248-fe2eaf94-3ede-4f7f-a527-997432366220.png">

Reply:
Does that mean defaultAreas should by default sorted?

Reply:
Yes, only sort areas if sorter exists.

Reply:
That means i have to pass a sorter prop?

Reply:
Let me clarify it by two examples.

If user pass an `areaSorter` to `ConfigProvider`, then sort the areas.
```jsx
<ConfigProvider
  locale={en}
  areaSorter={(a, b) => {
    if (a.name && b.name) {
      const nameA = a.name.toUpperCase();
      const nameB = b.name.toUpperCase();
      if (nameA < nameB) {
        return -1;
      }
      if (nameA > nameB) {
        return 1;
      }
      return 0;
    }
    return 0;
  }}
>
  <CountryPhoneInput />
</ConfigProvider>
```

If `areaSorter` is not set, then use `defaultAreas` directly. Don't change the order defined by source code.
```jsx
<ConfigProvider locale={zh}>
  <CountryPhoneInput />
</ConfigProvider>
```

Reply:
Oh I see that means i have to capture the if condition that checks if areaSorter is set on config.tsx?

Reply:
That's correct. Just add an if condition after the `areas` which is processed by `areaFilter` & `areaMapper`. If user provide the `areaFilter`, then use it.

Reply:
Then I think `defaultAreaSorter` is no need anymore.

Reply:
Hehehehe you know its my first time writing typescript thats why I keep on asking more questions.

Reply:
Hi @helsonxiao I made the necessary changes please review. Thank you.

Reply:
Sorry, maybe some of my words mislead you.

This type is required.
```ts
export type AreaSorter = (a: Area, b: Area) => number;
// areaSorter?: AreaSorter;
```

All we need to do is using sorter when it exists. This will keep areas order as we expected.

```js
let processedAreas = areas
  .filter((area, index, array) => areaFilter(area, index, array))
  .map((area, index, array) => areaMapper(area, index, array));

if (areaSorter) {
  processedAreas = processedAreas.sort(...);
}
```

Reply:
Does that mean the user has to pass sort logic on configProvider:

 areaSorter={(a, b) => {
    if (a.name && b.name) {
      const nameA = a.name.toUpperCase();
      const nameB = b.name.toUpperCase();
      if (nameA < nameB) {
        return -1;
      }
      if (nameA > nameB) {
        return 1;
      }
      return 0;
    }
    return 0;
  }}

Reply:
Yes. Here's the usage.

> Let me clarify it by two examples.
> 
> If user pass an `areaSorter` to `ConfigProvider`, then sort the areas.
> 
> ```js
> <ConfigProvider
>   locale={en}
>   areaSorter={(a, b) => {
>     if (a.name && b.name) {
>       const nameA = a.name.toUpperCase();
>       const nameB = b.name.toUpperCase();
>       if (nameA < nameB) {
>         return -1;
>       }
>       if (nameA > nameB) {
>         return 1;
>       }
>       return 0;
>     }
>     return 0;
>   }}
> >
>   <CountryPhoneInput />
> </ConfigProvider>
> ```
> 
> If `areaSorter` is not set, then use `defaultAreas` directly. Don't change the order defined by source code.
> 
> ```js
> <ConfigProvider locale={zh}>
>   <CountryPhoneInput />
> </ConfigProvider>
> ```

Reply:
@helsonxiao please check now.

Reply:
LGTM now